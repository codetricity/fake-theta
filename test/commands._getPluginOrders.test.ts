import { request, toMatchInvalidHeaderParameterError } from './helper';

describe('POST /osc/commands/execute _getPluginOrders', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera._getPluginOrders' })
      .expect(200);
    expect(res.body.results.pluginOrders).toMatchSnapshot();
  });
});

describe('THETA X POST /osc/commands/execute _getPluginOrders', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera._getPluginOrders' })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.body.results.pluginOrders).toMatchSnapshot();
  });
});

describe('THETA Z1 /osc/commands/execute _getPluginOrders', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera._getPluginOrders' })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.body.results.pluginOrders).toMatchSnapshot();
  });
});

describe('Unknown Type POST /osc/commands/execute _getPluginOrders', () => {
  it('should respond invalidHeaderParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera._getPluginOrders' })
      .set('emulating-theta-model', 'unknown')
      .expect(400);
    toMatchInvalidHeaderParameterError(res);
  });
});
