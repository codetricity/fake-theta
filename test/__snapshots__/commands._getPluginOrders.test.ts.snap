// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /osc/commands/execute _getPluginOrders should respond with results 1`] = `
[
  "com.example.mytestapplication1",
  "com.example.mytestapplication2",
  "com.example.mytestapplication3",
  "com.example.mytestapplication4",
  "com.example.mytestapplication5",
]
`;

exports[`THETA X POST /osc/commands/execute _getPluginOrders should respond with results 1`] = `
[
  "com.example.mytestapplication1",
  "com.example.mytestapplication2",
  "com.example.mytestapplication3",
  "com.example.mytestapplication4",
  "com.example.mytestapplication5",
]
`;

exports[`THETA Z1 /osc/commands/execute _getPluginOrders should respond with results 1`] = `
[
  "com.theta.remoteplayback",
  "com.theta360.remotecontrol",
  "com.theta360.usbstorage",
]
`;
